driver.go

```diff
@@ -632,35 +618,85 @@ func memoryInBytes(strmem string) (int64, error) {
 	}
 }
 
-func parseImage(image string) (string, error) {
-	// strip http/https and docker transport prefix
-	for _, prefix := range []string{"http://", "https://", "docker://"} {
-		if strings.HasPrefix(image, prefix) {
-			image = strings.Replace(image, prefix, "", 1)
+// Creates the requested image if missing from storage
+// returns the 64-byte image ID as an unique image identifier
+func (d *Driver) createImage(image string) (string, error) {
+	var imageID string
+	imageName := image
+	// If it is a shortname, we should not have to worry
+	// Let podman deal with it according to user configuration
+	if !shortnames.IsShortName(image) {
+		imageRef, err := parseImage(image)
+		if err != nil {
+			return imageID, fmt.Errorf("invalid image reference %s: %w\n", image, err)
```

Comment:
Why the new line?
```suggestion
			return imageID, fmt.Errorf("invalid image reference %s: %w", image, err)
```

Reply:
This is me sometimes thinking in C by default...